search/search-domain/src/main/java/io/camunda/search/filter/UserFilter.java

```diff
@@ -8,17 +8,29 @@
 package io.camunda.search.filter;
 
 import io.camunda.util.ObjectBuilder;
+import java.util.HashSet;
+import java.util.Set;
 
-public record UserFilter(Long key, String username, String name, String email)
+public record UserFilter(Set<Long> keys, String username, String name, String email)
     implements FilterBase {
   public static final class Builder implements ObjectBuilder<UserFilter> {
-    private Long key;
+    private Set<Long> keys = new HashSet<>();
     private String username;
     private String name;
     private String email;
 
+    @Deprecated
     public Builder key(final Long value) {
-      key = value;
+      if (value != null) {
+        keys = Set.of(value);
+      } else {
+        keys = new HashSet<>();
```

Comment:
```suggestion
        keys = Set.of();
```
I generally prefer that for creating empty sets as it returns an immutable set.